street_network/src/transform/classify_intersections.rs

```diff
@@ -34,8 +33,50 @@ fn guess_complexity(_inter: &Intersection, roads: Vec<OriginalRoad>) -> Intersec
         return Connection;
     }
 
-    // A MultiConnection is characterised by exactly two connected dividing lines (lines that
-    // separate traffic in different directions) and no traffic that crosses them.
+    // A MultiConnection is characterised by exactly one dividing line traveling through it (the
+    // line that separates traffic in different directions) and no traffic that crosses it.
+    if roads.len() == 3 {
+        let mut num_roads_in = 0;
+        let mut num_roads_out = 0;
+        let mut num_roads_inout = 0;
+        for or in roads {
+            let is_outward = or.i1 == *id;
+            let road = streets.roads.get(&or).unwrap();
+            match road.oneway_for_driving() {
+                Some(Direction::Fwd) => {
+                    if is_outward {
+                        num_roads_out += 1
+                    } else {
+                        num_roads_in += 1
+                    }
+                }
+
+                Some(Direction::Back) => {
+                    if is_outward {
+                        num_roads_in += 1
+                    } else {
+                        num_roads_out += 1
+                    }
+                }
+                None => num_roads_inout += 1,
+            }
+        }
+
+        if num_roads_inout == 0 {
+            // The simplest MultiConnect
+            return MultiConnection;
+        }
+
+        // detect the simple case of a single road splitting
+        if num_roads_inout == 1 {
+            // TODO Determine if it is possible to turn from the 'in' road to the 'out' road.
```

Comment:
![Screenshot from 2022-09-20 11-46-44](https://user-images.githubusercontent.com/1664407/191163013-a0c6950e-d740-41fa-836a-67a4e991c90d.png)
`aurora_sausage_link` is maybe a good simple example to look at...

![pxl_20220920_035003961_720](https://user-images.githubusercontent.com/1664407/191163538-848c643d-4230-41ae-935d-2982e305d9c3.jpg)
These are the movements possible, just from the cartesian product of every inbound road going to every outbound. In the T intersection case, two of the movements conflict, because they have the same destination (the south piece of road). Don't need any geometry to decide that.

In the dual carriageway split case, there's just a northbound or southbound movement. There is a U-turn, but it's explicitly banned in OSM, so this should show up in the `turn_restrictions: Vec<(RestrictionType, OriginalRoad)>` list.

Reply:
Yes, I had found this example too, which is great for having both cases right next to each other!

I was going to wait until Movements were represented in some consistent way before progressing on this, but I will keep going using just `turn_restrictions` as a stand in for the full movements that can be fleshed out later.